File: ios/ExpoAdaptiveGlassMagnifierModule.swift
import ExpoModulesCore

public class ExpoAdaptiveGlassMagnifierModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoAdaptiveGlassMagnifier")

    View(GlassMagnifierView.self) {
      Prop("lensWidth") { (view, value: Double) in view.lensWidth = CGFloat(value) }
      Prop("lensHeight") { (view, value: Double) in view.lensHeight = CGFloat(value) }
      Prop("magnification") { (view, value: Double) in view.magnification = CGFloat(value) }
      Prop("lift") { (view, value: Double) in view.lift = CGFloat(value) }
      Prop("refraction") { (view, value: Bool) in view.refraction = value }
      Prop("disabled") { (view, value: Bool) in view.disabled = value }

      OnViewDidUpdateProps { view in
        view.propsDidUpdate()
      }
    }
  }
}

File: android/src/main/java/expo/modules/adaptiveglass/ExpoAdaptiveGlassMagnifierModule.kt
package expo.modules.adaptiveglass

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAdaptiveGlassMagnifierModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoAdaptiveGlassMagnifier")

    View(GlassMagnifierView::class) {
      Prop("lensWidth") { view: GlassMagnifierView, value: Float -> view.lensWidth = value }
      Prop("lensHeight") { view: GlassMagnifierView, value: Float -> view.lensHeight = value }
      Prop("magnification") { view: GlassMagnifierView, value: Float -> view.magnification = value }
      Prop("lift") { view: GlassMagnifierView, value: Float -> view.lift = value }
      Prop("refraction") { view: GlassMagnifierView, value: Boolean -> view.refraction = value }
      Prop("disabled") { view: GlassMagnifierView, value: Boolean -> view.disabled = value }
    }
  }
}
